text/0000-route-model-name-property.md

```diff
@@ -0,0 +1,149 @@
+- Start Date: 2019-02-01
+- Relevant Team(s): (fill this in with the [team(s)](README.md#relevant-teams) to which this RFC applies)
+- RFC PR: (after opening the RFC PR, update this with a link to it and update the file name)
+- Tracking:
+
+# Route Model Name Property
+
+## Summary
+
+Add a property, `modelName`, to the route that will be the key that the model hook's return value is assigned to in the controller.
+
+## Motivation
+
+The default key for the routes model hook return on the controller is `model`. This name can be vague. The name is only expressive in the case where the route/controller naming represents the resource, e.g. `foo` route returns a `foo` model instance, and relies on the developer being aware of the convention, and for no other developer to ignore (circumvent?) the convention and return something else in the model hook. By adding a `modelName` property we allow developer to be expressive, and it will help future developers understand explicitly in the controller what type of model they are working with without having to refer back to the model hook.
+
+Additionally it adds some context to the current behaviour of assigning the model hook return to the controller, which is not transparent
+
+## Detailed design
+
+Currently if a developer wants to assign a custom name for the routes model value in the controller they must do this:
+
+```
+app/routes/posts.js
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model() {
+    return this.store.findAll('post');
+  },
+  setupController(model, controller){
+    controller.set('articles', model);
+  }
+});
+```
+
+This RFC proposes a  solution like this:
+
+```
+app/routes/posts.js
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model() {
+    return this.store.findAll('post');
+  },
+  modelName: articles
+});
+```
+
+- Add a property called `modelName` to the Route API. When the controller is initialised, the return value from the routes `model()` hook is assigned to the controller with value of the `modelName` property as the key.
+- This change would not require us to stop assigning the return value to `model` key on the controller as well.
+- Generator could be updated so that `ember generate route foo` would add `modelName: foo` to the generated route file.
```

Comment:
Ideally this wouldn't be necessary and `modelName` would internally be set by Ember itself to the right thing

that however would be a breaking change, as it might overwrite things on the controller that were previously not set 🤔

Reply:
So would the route name be the default value for `modelName`?